### PR TITLE
[SPARK-46950][CORE][SQL] Align `not available codec` error-class

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to align `not available codec` error-class, includes:
- In `core`, convert `CODEC_NOT_AVAILABLE` to `CODEC_NOT_AVAILABLE.WITH_CONF_SUGGESTION`.
- In `sql`, use `CODEC_NOT_AVAILABLE.WITH_AVAILABLE_CODECS_SUGGESTION` to cover `IllegalArgumentException`.

### Why are the changes needed?
When I was solving a case where `the compression option` is `null`, as follows:
image
I found that we could first complete the extraction logic for `the codec error`. At the same time, I found that there was already an `CODEC_NOT_AVAILABLE` error in the `error-classes.json`, but it had some differences from the error prompt in `SQL` for compress. Therefore, I proposed using `CODEC_NOT_AVAILABLE` as `a parent class error` and adding `two subclass errors`: `WITH_AVAILABLE_CODECS_SUGGESTION` and `WITH_CONF_SUGGESTION`.

### Does this PR introduce _any_ user-facing change?
Yes.

### How was this patch tested?
- Update existed UT.
- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
image

**Comment by panbingkun** [comment]
After it, I will continue to resolve the case where `the compression option` is `null`.

**Comment by panbingkun** [comment]
cc @MaxGekk

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetOptions.scala [hunk 1]
```diff
@@ -55,8 +56,8 @@ class ParquetOptions(
     if (!shortParquetCompressionCodecNames.contains(codecName)) {
       val availableCodecs =
         shortParquetCompressionCodecNames.keys.map(_.toLowerCase(Locale.ROOT))
-      throw new IllegalArgumentException(s"Codec [$codecName] " +
-        s"is not available. Available codecs are ${availableCodecs.mkString(", ")}.")
+      throw QueryExecutionErrors.codecNotAvailableError(
+          codecName, availableCodecs.mkString(", "))
```
Comment (MaxGekk) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetOptions.scala:60]: Adjust indentation here.

**Review by MaxGekk — APPROVED** [review]
LGTM except of a minor comment.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetOptions.scala (hunk 1 above)
Comment (MaxGekk) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetOptions.scala:60]:
```suggestion
        codecName, availableCodecs.mkString(", "))
```

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**